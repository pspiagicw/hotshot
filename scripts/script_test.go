package tests

import (
	"io/fs"
	"os/exec"
	"path/filepath"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

type scriptTest struct {
	name string
	file string
}

func checkFile(t *testing.T, filename string) {
	cmd := exec.Command("../hotshot", filename)
	output, err := cmd.Output()
	if err != nil {
		t.Errorf("Error running interpreter! %v", err)
		t.Log(string(output))
	} else {
		snaps.MatchSnapshot(t, string(output))
	}
}

func TestScripts(t *testing.T) {
	tt := getFiles()

	for _, test := range tt {
		t.Run(test.name, func(t *testing.T) {
			checkFile(t, test.file)
		})
	}

}
func getFiles() []scriptTest {
	files := []scriptTest{}
	filepath.WalkDir("files", func(path string, d fs.DirEntry, err error) error {
		if err != nil {
			return err
		}
		if !d.IsDir() {
			files = append(files, scriptTest{name: d.Name(), file: path})
		}

		return nil
	})

	return files

}
