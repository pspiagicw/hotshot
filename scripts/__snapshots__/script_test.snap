
[TestScripts/Hello_World - 1]
 Hello, World 

---

[TestScripts/Data_Types - 1]
 2 
 this is a string 
 true 
 false 

---

[TestScripts/Arithmetic_Types - 1]
 3 
 17 
 1 
 14 

---

[TestScripts/Booleans - 1]
 true 
 false 
 true 
 false 
 false 
 true 
 false 
 false 
 false 
 true 
 true 
 true 
 false 
 true 
 true 
 false 

---

[TestScripts/arithmetic.ht - 1]
 3 
 17 
 1 
 14 

---

[TestScripts/booleans.ht - 1]
 true 
 false 
 true 
 false 
 false 
 true 
 false 
 false 
 false 
 true 
 true 
 true 
 false 
 true 
 true 
 false 

---

[TestScripts/data-types.ht - 1]
 2 
 this is a string 
 true 
 false 

---

[TestScripts/factorial.ht - 1]
 120 
 3628800 

---

[TestScripts/fibonacci.ht - 1]
 11 
 47 

---

[TestScripts/fizzbuzz.ht - 1]
 [!!] Error parsing the file.
Expected a token for a statement, found: Token(Type: MINUS, Value: '-')
Expected a token for a statement, found: Token(Type: MINUS, Value: '-')
 [!!] Not executing till errors resolved!

---

[TestScripts/hello-world.ht - 1]
 Hello, World 

---

[TestScripts/sum-of-digits.ht - 1]
 1 
 10 
 15 

---
